Refactor RingLibSDL project for Android

ysdragon opened the pull request:
I have refactored the RingLibSDL Android project as I mentioned before.

Please test the new project.

I have commented out the `fexists` usage in GameEngine because fexists always returns `0`. I believe this is because it doesn't have access to the assets directory, and it was causing the game to crash on Android.

This project has been tested on Windows and Linux.

Additionally, the generated .apk file works as expected on Android 5 to Android 15 (tested on two real devices).

MahmoudFayed commented:
Hello Youssef

Thank you very much for the updates

1- I like the good documentation & automation scripts 
2- With respect to fexists(), we can customize/enable/disable its usage using isAndroid() function: https://ring-lang.github.io/doc1.23/systemfunc.html#isandroid-function 
3- I will do more tests during this week 

Greetings,
Mahmoud

ysdragon commented:
Hello Mahmoud,

> I have already updated Ring Website - News Section
Added: Building Games for Android
A link to this page: https://github.com/ring-lang/ring/tree/master/extensions/android/ringlibsdl

Thank you for adding and highlighting it in the news section!

> Feel free to share the link in Ring Group, highlighting the important changes (Using recent versions of libraries, automated scripts for installing Android SDK, testing using multiple devices) - And a screen shot for one of your animation samples

Of course, I will share that.

Also, I will add support for macOS to the automated scripts later.

Best regards,
Youssef

MahmoudFayed commented:
Hello Youssef

You are welcome 

Greetings,
Mahmoud
